text/0000-katello-as-a-plugin.md

```diff
@@ -0,0 +1,107 @@
+- Feature Name: Katello as a (Real) Plugin
+- Start Date: 2016-09-09
+- RFC PR: (leave this empty)
+- Issue: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+The goal of this effort is to allow Katello to be treated from a community and technical perspective as any other plugin.
+
+# Motivation
+[motivation]: #motivation
+
+Katello was originally a stand alone Rails application that attempted to integrate with Foreman through a single-sign-on process, Rails engines that knew how to communicate with one another and switching menu structures. After a desire to centralize on a single community and bring integration tighter, Katello was moved to be a plugin but given the constraints on Katello for depoyment along with timeline the deployment of Foreman and Katello was made unique when compared to any other plugin. 
+
+# Detailed design
+[design]: #detailed-design
+
+Given this is a large area of work with many facets, the design will list out each issue with proposed design solutions and any drawbacks. This may be able to be split out to their own RFCs over time.
+
+## Certificates
+
+Foreman defaults to using Puppet certificates for the web UI. Katello requires a certificate authority to hand to Candlepin for cutting client certificates and entitlement certificates. The same certificate authority used to cut client certificates is needed to have signed the server certificates so that clients can talk back to the server and access HTTPs based repositories. Further, certificates are needed to configure communication with Pulp, Qpid, Capsules (Smart Proxies) and creating entities like trustores for Candlepin.
+
+Another aspect of certificates is the desire by users to be able to supply their own certificates for the web UI. Users should be able to change those fluidly without having to propagate new certificates across their entire client infrastructure.
```

Comment:
Or we can install multiple trusted CA certs. User could provide extra CA cert that would be installed with both Puppet and Candlepin CA certs. This would allow us to easily start using new CA that would be used if users do not provide their own. This would be useful today in scenario where I do not use Puppet nor Candlepin but I still want to have x509 certs.